components/script/script_runtime.rs

```diff
@@ -915,11 +915,19 @@ unsafe fn set_gc_zeal_options(_: *mut RawJSContext) {}
 /// enabling an interrupt to be requested
 /// from a thread other than the one running JS using that context.
 #[derive(Clone)]
-pub struct ContextForRequestInterrupt(Arc<Mutex<Option<*mut RawJSContext>>>);
+pub struct ContextForRequestInterrupt(Arc<Mutex<Option<ThreadSafeContext>>>);
```

Comment:
I am also not sure about this, as safety of sending JSContext is from mutex IIUC, and generally JSContext is not thread safe, so `ThreadSafeContext` is easily completely wrong. My feeling is that clippy gives bad warnings for our safe wrapper, so it might be better to just allow some of them.

Reply:
noted!!

Reply:
cc @jdm